website_portal_purchase/views/orders.xml

```diff
@@ -8,7 +8,7 @@
               customize_show="True"
               active="True">
         <xpath expr="//t[@name='supplier_menu']">
-            <li>
+            <li t-att-class="'active' if request.httprequest.path.startswith('/my/purchase/orders') else ''">
```

Comment:
You can use `request.httprequest.path.startswith('/my/purchase/orders') and 'active'`. In wrong cases, it will be evaluated to `False` and server-side qweb engine AFAIK plainly removes `t-att-*` attributes that are `False`.

Just a suggestion, not requirement.